packages/react-button/src/components/CompoundButton/useCompoundButtonStyles.ts

```diff
@@ -1,356 +1,224 @@
-        },
-      },
-    };
-  },
-  rootDisabled: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
-
-    return {
+    ':hover': {
       [`& .${CompoundButtonClassNames.secondaryContent}`]: {
-        color: compoundButtonTokens.disabled?.secondaryContentColor,
-      },
-
-      ':hover': {
-        [`& .${CompoundButtonClassNames.secondaryContent}`]: {
-          color: compoundButtonTokens.disabled?.hovered?.secondaryContentColor,
-        },
+        color: theme.alias.color.neutral.neutralForegroundDisabled,
       },
+    },
 
-      ':active': {
-        [`& .${CompoundButtonClassNames.secondaryContent}`]: {
-          color: compoundButtonTokens.disabled?.pressed?.secondaryContentColor,
-        },
+    ':active': {
+      [`& .${CompoundButtonClassNames.secondaryContent}`]: {
+        color: theme.alias.color.neutral.neutralForegroundDisabled,
       },
-    };
-  },
-  rootSmall: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
-
-    return {
-      padding: `${compoundButtonTokens.small?.paddingY} ${compoundButtonTokens.small?.paddingX}`,
-    };
-  },
-  rootLarge: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+    },
+  }),
+});
 
-    return {
-      padding: `${compoundButtonTokens.large?.paddingY} ${compoundButtonTokens.large?.paddingX}`,
-    };
+const useRootIconOnlyStyles = makeStyles({
+  small: {
+    maxWidth: '48px',
+    minWidth: '48px',
   },
-  rootIconOnly: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
-
-    return {
-      maxWidth: compoundButtonTokens.iconOnly?.maxWidth,
-      minWidth: compoundButtonTokens.iconOnly?.minWidth,
-    };
+  medium: {
+    maxWidth: '52px',
+    minWidth: '52px',
   },
-  rootIconOnlySmall: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
-
-    return {
-      maxWidth: compoundButtonTokens.iconOnlySmall?.maxWidth,
-      minWidth: compoundButtonTokens.iconOnlySmall?.minWidth,
-    };
+  large: {
+    maxWidth: '56px',
+    minWidth: '56px',
   },
-  rootIconOnlyLarge: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+});
 
-    return {
-      maxWidth: compoundButtonTokens.iconOnlyLarge?.maxWidth,
-      minWidth: compoundButtonTokens.iconOnlyLarge?.minWidth,
-    };
-  },
-  childrenSmall: theme => {
-    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+const useChildrenStyles = makeStyles({
+  small: theme => ({
+    fontSize: theme.global.type.fontSizes.base[300],
+    lineHeight: theme.global.type.lineHeights.base[300],
+  }),
+  medium: theme => ({
+    fontSize: theme.global.type.fontSizes.base[300],
+    lineHeight: theme.global.type.lineHeights.base[300],
+  }),
+  large: theme => ({
+    fontSize: theme.global.type.fontSizes.base[400],
+    lineHeight: theme.global.type.lineHeights.base[400],
+  }),
+});
```

Comment:
It looks like `small` is the only style that's different from useButtonStyles here; just wanted to double check that that is correct. Also, useButtonStyles sets the font weight for these -- do you need to override the font weight for any of them?

Reply:
That is correct according to the design spec, just wanted to use the same pattern to be able to use `styles[state.size]` below. `fontWeight` is not changed on any of them according to the design spec.